src/gui/mzroll/infodialog.cpp

```diff
@@ -0,0 +1,81 @@
+#include "common/mixpanel.h"
+#include "infodialog.h"
+#include "mainwindow.h"
+#include "ui_infodialog.h"
+
+InfoDialog::InfoDialog(MainWindow* parent) :
+    QDialog(parent),
+    ui(new Ui::InfoDialog)
+{
+    ui->setupUi(this);
+    layout()->setSizeConstraint(QLayout::SetFixedSize);
+
+    auto tracker = parent->getUsageTracker();
+    auto name = tracker->userAttribute("Name").toString();
+    auto email = tracker->userAttribute("Email").toString();
+    if (email != "") {
+        ui->nameEntry->setText(name);
+        ui->emailEntry->setText(email);
+        ui->nameEntry->setDisabled(true);
+        ui->emailEntry->setDisabled(true);
+        ui->signInButton->setDefault(false);
+        ui->signInButton->setText("Sign out");
+    }
+
+    connect(ui->signInButton, &QPushButton::clicked, [this, tracker] {
+        if (ui->signInButton->text() == "Sign out") {
+            tracker->updateUser("Name", "");
+            tracker->updateUser("Email", "");
+            ui->nameEntry->clear();
+            ui->emailEntry->clear();
+            ui->nameEntry->setEnabled(true);
+            ui->emailEntry->setEnabled(true);
+            ui->signInButton->setDefault(true);
+            ui->signInButton->setText("Sign in");
+            return;
+        }
+
+        auto name = ui->nameEntry->text();
+        auto email = ui->emailEntry->text();
+        QRegularExpression emailPattern("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~"
+                                        "-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]"
+                                        "{0,61}[a-zA-Z0-9])?(?:\\.[a-zA"
+                                        "-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-"
+                                        "zA-Z0-9])?)*$");
+        auto match = emailPattern.match(email);
+        if (match.hasMatch()) {
```

Comment:
If the pattern does not match, show a message saying "Add a valid email ID" or something.

Reply:
Added a label for informing user in three different scenarios (sign-in, sign-out and invalid email).